atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/atomic/MarkAndCasConsensusForgettingStore.java

```diff
@@ -66,30 +86,23 @@ public void mark(Set<Cell> cells) {
      */
     @Override
     public void atomicUpdate(Cell cell, byte[] value) throws CheckAndSetException {
-        CheckAndSetRequest request = CheckAndSetRequest.singleCell(tableRef, cell, inProgressMarker, value);
-        kvs.checkAndSet(request);
+        autobatcher.apply(ImmutableCasRequest.of(cell, inProgressMarkerBuffer, ByteBuffer.wrap(value)));
     }
 
     @Override
     public void atomicUpdate(Map<Cell, byte[]> values) throws MultiCheckAndSetException {
-        byte[] row = getRowName(values);
-        Map<Cell, byte[]> expected =
-                values.keySet().stream().collect(Collectors.toMap(cell -> cell, _ignore -> inProgressMarker));
-        MultiCheckAndSetRequest request = MultiCheckAndSetRequest.multipleCells(tableRef, row, expected, values);
-        kvs.multiCheckAndSet(request);
+        values.forEach(this::atomicUpdate);
```

Comment:
This is technically not atomic as it can be batched in multiple batches. For our purpose this is good enough, but then it needs to be documented at the interface level. Otherwise the autobatcher must accept a collection instead